flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResourceManagerPartitionTracker.java

```diff
@@ -64,4 +66,12 @@ void processTaskExecutorClusterPartitionReport(
      * @return tracked datasets
      */
     Map<IntermediateDataSetID, DataSetMetaInfo> listDataSets();
+
+    /**
+     * Returns all the shuffle descriptors of cluster partitions for the intermediate dataset.
+     *
+     * @param dataSetID The id of the intermediate dataset.
+     * @return the shuffle descriptors.
+     */
+    List<ShuffleDescriptor> getClusterPartitionShuffleDescriptors(IntermediateDataSetID dataSetID);
```

Comment:
Does we call this in the rpc main thread? I am asking because I am wondering if there is any race condition.